song_player/SongPlayer.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <QAbstractButton>
+#include <QBoxLayout>
+#include <QCoreApplication>
+#include <QDebug>
+#include <QFileDialog>
+#include <QLabel>
+#include <QMediaPlayer>
+#include <QMouseEvent>
+#include <QShortcut>
+#include <QSlider>
+#include <QStandardPaths>
+#include <QStyle>
+#include <QTime>
+#include <QToolButton>
+#include <QWidget>
+
+class SongPlayer : public QWidget
+{
+    Q_OBJECT
+public:
+    SongPlayer(QWidget *parent = 0);
+
+public slots:
+    void openFile();
```

Comment:
This class does a shitload. The song logic and the UI needs to be separated into different classes.